datalad/interface/crawl_init.py

```diff
@@ -0,0 +1,117 @@
+
+    Examples:
+
+    $ datalad crawl-init \
+        --template openfmri \
+        --func superdataset_pipeline
+
+    $ datalad crawl-init \
+        --template fcptable \
+        dataset=Baltimore tarballs=True
+    """
+    _params_ = dict(
+        template=Parameter(
+            args=("-t", "--template"),
+            action="store",
+            constraints=EnsureStr() | EnsureNone(),
+            doc="""flag if template is specified by user"""),
+        template_func=Parameter(
+            args=("-f", "--func"),
+            action="store",
+            doc="""flag if function is specified by user"""),
+        args=Parameter(
+            args=("args",),
+            nargs="*",
+            type=OrderedDict or list,
+            doc="""keyword arguments to pass into the template function generating actual pipeline,
+            organized in an ordered dict"""),
+        commit=Parameter(
+            args=("--commit",),
+            action="store_true",
+            doc="""flag is user wants to commit file into git repo"""),
+    )
+
+    @staticmethod
+    def __call__(args=None, template=None, template_func=None, commit=False):
+
+        lgr.debug("Creating crawler configuration for template %s under %s",
+                  template)
+
+        crawl_config_dir = opj(curdir, CRAWLER_META_DIR)
+        if not exists(crawl_config_dir):
+            lgr.log(2, "Creating %s", crawl_config_dir)
+            makedirs(crawl_config_dir)
+
+        crawl_config = opj(crawl_config_dir, CRAWLER_META_CONFIG_FILENAME)
+        crawl_config_repo_path = opj(CRAWLER_META_DIR, CRAWLER_META_CONFIG_FILENAME)
+        cfg_ = SafeConfigParserWithIncludes()
+        cfg_.add_section(CRAWLER_PIPELINE_SECTION)
+
+        if template:
+            cfg_.set(CRAWLER_PIPELINE_SECTION, 'template', template)
+
+        if template_func:
+            cfg_.set(CRAWLER_PIPELINE_SECTION, 'func', template_func)
+
+        if args:
+            if isinstance(args, list):
+                newargs = dict
+                for item in args:
+                    k, v = item.split('=', 1)
+                    newargs[k] = v
+                args = newargs
+            elif isinstance(args, dict):
+                pass
+            else:
+                t = type(args)
+                raise ValueError("args entered must be given in a list or dict, were given as %s", t)
+            args = OrderedDict(sorted(args.items()))
+            for k, v in args.items():
+                cfg_.set(CRAWLER_PIPELINE_SECTION, "_" + k, str(v))
+
+        with open(crawl_config, 'w') as f:
+            cfg_.write(f)
+
+        if commit:
+            repo = GitRepo(curdir)
+            repo.add(crawl_config_repo_path)
+            if repo.dirty:
+                repo.commit("Initialized crawling configuration to use template %s" % template)
+            else:
+                lgr.debug("Repository is not dirty -- not committing")
```

Comment:
this all to be gone...   
overall advice -- do not copy/paste code almost ever -- duplication is evil.  If there is a function -- use it.  If it doesn't exactly fit your needs, generalize it so it does fit old usecase and new one. In this case it doesn't need any customization